### PR TITLE
Refactor external command executor

### PR DESCRIPTION
**Pull request by izumin5210**

### CODE REVIEW

File: pkg/command/executor.go [hunk 1]
```diff
@@ -84,16 +90,16 @@ func (c *command) Exec() (out []byte, err error) {
 		wg.Add(2)
 		go func() {
 			defer wg.Done()
-			io.Copy(c.outWriter, io.TeeReader(outReader, &buf))
+			io.Copy(e.outWriter, io.TeeReader(outReader, &buf))
```
Comment (watchbunny) [inline pkg/command/executor.go]: **[errcheck]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
	io.Copy(e.outWriter, io.TeeReader(outReader, &buf))

File: pkg/command/executor.go [hunk 2]
```diff
@@ -84,16 +90,16 @@ func (c *command) Exec() (out []byte, err error) {
...
 		}()
 		closers = append(closers, outReader.Close)
 		go func() {
 			defer wg.Done()
-			io.Copy(c.errWriter, io.TeeReader(errReader, &buf))
+			io.Copy(e.errWriter, io.TeeReader(errReader, &buf))
```
Comment (watchbunny) [inline pkg/command/executor.go]: **[errcheck]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
	io.Copy(e.errWriter, io.TeeReader(errReader, &buf))

File: pkg/command/executor.go [hunk 3]
```diff
@@ -61,8 +58,17 @@ func (c *command) Exec() (out []byte, err error) {
...
+	out, err = e.exec(ctx, c, cmd)
+
+	signal.Reset()
+	close(sigCh)
+
+	wg.Wait()
+	return
+}
+
+func (e *executor) exec(ctx context.Context, c *Command, cmd *exec.Cmd) (out []byte, err error) {
```
Comment (watchbunny) [inline pkg/command/executor.go]: **[unparam]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
(*executor).exec - ctx is unused

File: pkg/command/executor.go [hunk 4]
```diff
@@ -61,8 +58,17 @@ func (c *command) Exec() (out []byte, err error) {
...
 
-	clog.Debug("execute", "command", cmd.Args, "dir", cmd.Dir)
-	if c.ioConnected {
+	out, err = e.exec(ctx, c, cmd)
+
+	signal.Reset()
+	close(sigCh)
+
+	wg.Wait()
+	return
```
Comment (watchbunny) [inline pkg/command/executor.go]: **[wraperr]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
	github.com/izumin5210/grapi/pkg/command.Exec

Comment (izumin5210) [reply pkg/command/executor.go]: fixed in 1e9fd43

File: pkg/command/executor.go (hunk 3 above)
Comment (izumin5210) [reply pkg/command/executor.go]: fixed in 1e9fd43

**Pull request merged**